lib/public/TaskProcessing/IManager.php

```diff
@@ -37,6 +37,14 @@ public function hasProviders(): bool;
 	 */
 	public function getProviders(): array;
 
+	/**
+	 * @param string $taskType
+	 * @return IProvider
+	 * @throws \OCP\TaskProcessing\Exception\Exception
+	 * @since 30.0.0
+	 */
+	public function getPreferredProvider(string $taskType);
```

Comment:
But you can't specify what you prefer?
It returns the first one matching the taskType?

```suggestion
	public function getFirstProviderForTaskType(string $taskType);
```

Reply:
It's a stand-in implementation until we implement the settings UI for the preferred providers